roles/network_plugin/calico/templates/calico-node.yml.j2

```diff
@@ -44,6 +44,29 @@ spec:
       terminationGracePeriodSeconds: 0
 {% if calico_version is version('v3.4.0', '>=') %}
       initContainers:
+{% if calico_datastore == "kdd" and calico_version is version('v3.6.0', '>=') %}
```

Comment:
Let's maybe have a separate trigger for this? Or to make sure it is upgrade not a fresh installation?

Reply:
This container does nothing on a fresh install; it acts only when an upgrade to calico-ipam is necessary

These are the logs I got from this init container on a fresh installation:
```
... [INFO][1] k8s.go 228: Using Calico IPAM
... [INFO][1] migrate.go 64: checking host-local IPAM data dir dir existence...
... [INFO][1] migrate.go 66: host-local IPAM data dir dir not found; no migration necessary, successfully exiting...
```
I would recommend keeping it this way since it corresponds to calico templates https://github.com/projectcalico/calico/blob/master/_includes/v3.6/charts/calico/templates/calico-node.yaml#L47

Reply:
Let's keep it this way. My concern was mostly about additional time it takes. I suppose it should be negligibly small here.